File: Java/2024_cursoemvideo_poo/aula02_classes/src/Caneta.java
public class Caneta {
    String modelo;
    String cor;
    float ponta;
    int carga;
    boolean tampada;
    void status() {
        System.out.println("A cor é "+this.cor);
        System.out.println("A ponta é "+this.ponta);
        if (this.tampada == true) {
            System.out.println("Ela está tampada");
        } else {
            System.out.println("Ela está destampada.");
        }

    }

    void rabiscar() {
        if (this.tampada == true) {
            System.out.println("Nã dá. A caneta está tampada.");
        } else {
            System.out.println("Rabisco feito.");
        }
    }

    void tampar() {
        this.tampada = true;
    }

    void destampar (){
        this.tampada = false;
    }
}

File: PHP/2016_cursoemvideo_poo/aula03_visibilidade/Caneta.php
<?php 
class Caneta {
    public $modelo;
    public $cor;
    private $ponta;
    protected $carga;
    private $tampada;
    
    public function rabiscar() {
        if ($this -> tampada == true) {
            echo "Função rabiscar(): Não dá. Tá tampada.<br><br>";
        } else {
            echo "Função rabiscar(): Rabisco feito.<br><br>";
        }
    }

    public function tampar() {
        $this -> tampada = true;
        echo "Função tampar(): a caneta foi tampada.<br><br>";
    }

    public function destampar() {
        $this -> tampada = false;
        echo "Função destampar(): a caneta foi destampada.<br><br>";
    }
}

?>